### PR TITLE
plugins/nvim-snippets: init

### PR DESCRIPTION
**Pull request by psfloyd**
Adds support for [nvim-snippets](https://github.com/garymjr/nvim-snippets).

### CODE REVIEW

File: tests/test-sources/plugins/snippets/nvim-snippets.nix
Comment (GaetanLepage) [inline tests/test-sources/plugins/snippets/nvim-snippets.nix:77]:
```suggestion
        action.__raw ''
          function()
```

File: plugins/snippets/nvim-snippets.nix
Comment (GaetanLepage) [inline plugins/snippets/nvim-snippets.nix:16]:
```suggestion
    create_autocmd = helpers.defaultNullOpts.mkBool false ''
      Optionally load all snippets when opening a file.
      Only needed if not using nvim-cmp.
    '';

```

Please, adopt the same coding style as the rest of the code base. This ensure consistency across the project.

**Review by GaetanLepage — APPROVED** [review]
Thanks for this contribution.
Your PR looks promising !

I made a few minor remarks.

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
mistake.

File: tests/test-sources/plugins/snippets/nvim-snippets.nix
Comment (GaetanLepage) [inline tests/test-sources/plugins/snippets/nvim-snippets.nix:30]:
```suggestion
    plugins = {
      friendly-snippets.enable = true;

      cmp = {
        enable = true;

        settings = {
          sources = [ { name = "snippets"; } ];
          mapping.__raw = "require('cmp').mapping.preset.insert()";
        };
      };
      nvim-snippets = {
        enable = true;
        settings = {
          create_autocmd = true;
          create_cmp_source = true;
          friendly_snippets = true;
          ignored_filetypes = [ "lua" ];
          extended_filetypes = {
            typescript = [ "javascript" ];
          };
          global_snippets = [ "all" ];
          search_paths = [ { __raw = "vim.fn.stdpath('config') .. '/snippets'"; } ];
```

File: tests/test-sources/plugins/snippets/nvim-snippets.nix
Comment (GaetanLepage) [inline tests/test-sources/plugins/snippets/nvim-snippets.nix:9]: Usually we have an `defaults` test case that strictly copies the upstream defaults and eventually an `example` test case where we set non-defaults value.

File: plugins/snippets/nvim-snippets.nix
Comment (GaetanLepage) [inline plugins/snippets/nvim-snippets.nix:20]:
```suggestion
    '';

    create_cmp_source = helpers.defaultNullOpts.mkBool true ''
```

Please, add a line break between each option declaration.

File: plugins/snippets/nvim-snippets.nix
Comment (GaetanLepage) [inline plugins/snippets/nvim-snippets.nix:8]:
```suggestion
}:
let
  inherit (lib) types;
  inherit (lib.nixvim) defaultNullOpts;
in
helpers.neovim-plugin.mkNeovimPlugin config {
```

This is a brand-new coding style that we are trying out.

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
A bit more nit-picking...

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
This looks good to me !
@nix-community/nixvim could you please double-check quickly ?

@psfloyd, please squash your commits so as to have two left:
- `maintainers: add psfloyd`
- `plugins/nvim-snippets: init`

File: plugins/snippets/nvim-snippets.nix
Comment (MattSturgeon) [inline plugins/snippets/nvim-snippets.nix:50]:
```suggestion
```

**Review by MattSturgeon — COMMENTED** [review]
One nit, otherwise LGTM once other feedback is addressed!

**Comment by psfloyd** [comment]
It's ready to merge from my part. Squashed the commits and fixed the unnedded newline.

**Review by MattSturgeon — APPROVED** [review]
LGTM, thanks!

**Review by GaetanLepage — APPROVED** [review]
Well done, thanks !

**Comment by GaetanLepage** [comment]
@mergifyio queue

**Pull request merged**
